src/test/regress/expected/multi_view.out

```diff
@@ -268,3 +268,514 @@ SELECT * FROM  lineitems_by_orderkey WHERE l_orderkey = 100;
+       6 | Sun Jan 19 06:09:39.900888 2014 |       3 |     908 |     688 |        
+      13 | Sun Jan 19 22:09:26.256209 2014 |       2 |     755 |     584 |        
+      39 | Wed Jan 15 05:46:51.48765 2014  |      14 |     657 |     137 |        
+      39 | Sun Jan 19 11:26:47.45937 2014  |      12 |     118 |     165 |        
+      44 | Wed Jan 15 14:23:52.532426 2014 |       8 |     204 |     735 |        
+      44 | Sun Jan 19 05:53:34.829093 2014 |       4 |     758 |     205 |        
+      46 | Mon Jan 13 20:39:11.211169 2014 |       0 |     235 |     475 |        
+      46 | Wed Jan 15 09:14:57.471944 2014 |       2 |     407 |     664 |        
+      50 | Sat Jan 11 11:07:13.089216 2014 |       6 |     292 |     425 |        
+      58 | Sun Jan 19 22:36:14.795396 2014 |       2 |      86 |     311 |        
+      66 | Tue Jan 14 20:16:31.219213 2014 |      14 |     347 |     655 |        
+      74 | Tue Jan 21 01:38:39.570986 2014 |       9 |     334 |     642 |        
+      86 | Sun Jan 19 06:18:51.466578 2014 |      14 |     712 |     490 |        
+      87 | Sat Jan 11 20:46:28.439073 2014 |       2 |     528 |     311 |        
+      90 | Sun Jan 12 21:37:30.778206 2014 |      11 |     458 |     377 |        
+     100 | Sun Jan 19 22:32:08.284043 2014 |       2 |     384 |     149 |        
+(18 rows)
+
+-- determine if a recent user has done a given event type or not
+SELECT ru.user_id, CASE WHEN et.user_id IS NULL THEN 'NO' ELSE 'YES' END as done_event
+	FROM recent_users ru LEFT JOIN events_table et ON(ru.user_id = et.user_id AND et.event_type = 625) ORDER BY 2 DESC, 1;
+ user_id | done_event 
+---------+------------
+       6 | YES
+      13 | NO
+      39 | NO
+      44 | NO
+      46 | NO
+      50 | NO
+      58 | NO
+      66 | NO
+      71 | NO
+      74 | NO
+      86 | NO
+      87 | NO
+      90 | NO
+     100 | NO
+(14 rows)
+
+-- create a select only view
+CREATE VIEW selected_users AS SELECT * FROM users_table WHERE value_1 >= 120 and value_1 <150;
+CREATE VIEW recent_selected_users AS SELECT su.* FROM selected_users su JOIN recent_users ru USING(user_id);
+SELECT user_id FROM recent_selected_users GROUP BY 1 ORDER BY 1;
+ user_id 
+---------
+       6
+      13
+      39
+      44
+      46
+      50
+      58
+      66
+      71
+      74
+      86
+      90
+(12 rows)
+
+-- this would be supported when we implement where partition_key in (subquery) support
+SELECT et.* FROM events_table et WHERE et.user_id IN (SELECT user_id FROM recent_selected_users);
+ERROR:  could not run distributed query with subquery outside the FROM clause
+HINT:  Consider using an equality filter on the distributed table's partition column.
+-- it is supported if it is a router query
+SELECT count(*) FROM events_table et WHERE et.user_id IN (SELECT user_id FROM recent_selected_users WHERE user_id = 90);
+ count 
+-------
+   109
+(1 row)
+
+-- expected this to work but it did not
+(SELECT user_id FROM recent_users) 
```

Comment:
Why do you expect this to work? The UNION is not even in a subquery (and inside a view)? But even wrapping that into a top level query wouldn't help because then the leaf queries would include subqueries.